kotlin/toggle-between-feature-request-modes/src/main/java/com/esri/arcgisruntime/sample/togglebetweenfeaturerequestmodes/MainActivity.kt

```diff
@@ -0,0 +1,242 @@
+import com.esri.arcgisruntime.layers.FeatureLayer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.BasemapStyle
+import com.esri.arcgisruntime.mapping.Viewpoint
+import com.esri.arcgisruntime.mapping.view.DrawStatus
+import com.esri.arcgisruntime.mapping.view.DrawStatusChangedEvent
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.sample.togglebetweenfeaturerequestmodes.databinding.ActivityMainBinding
+import java.util.Collections
+import java.util.concurrent.atomic.AtomicInteger
+
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG: String = MainActivity::class.java.simpleName
+
+    private val activityMainBinding by lazy {
+        ActivityMainBinding.inflate(layoutInflater)
+    }
+
+    private val mapView: MapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    private val modeTV: TextView by lazy {
+        activityMainBinding.modeTV
+    }
+
+    private val populateTV: TextView by lazy {
+        activityMainBinding.populateTV
+    }
+
+    private val progressBar: ProgressBar by lazy {
+        activityMainBinding.progressBar
+    }
+
+    private val labelTV: TextView by lazy {
+        activityMainBinding.labelText
+    }
+
+    // instance of the FeatureLayer to be used by the map
+    private var featureLayer: FeatureLayer? = null
+
+    // currently selected feature mode: 0.Cache, 1.No cache, 2.Manual cache
+    private var featureModeSelected: Int = 0
+
+    // instance of the service feature table of street trees in Portland
+    private var featureTable: ServiceFeatureTable? =
+        ServiceFeatureTable("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/rest/services/Trees_of_Portland/FeatureServer/0")
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(activityMainBinding.root)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY)
+
+        mapView.apply {
+            // set the map to be displayed in the layout's MapView
+            this.map = ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC)
+            // set the starting viewpoint for the map view
+            setViewpoint(Viewpoint(45.5266, -122.6219, 6000.0))
+            // show a progress indicator when the map view is drawing (e.g. when fetching caches)
+            addDrawStatusChangedListener { e: DrawStatusChangedEvent ->
+                // true if DrawStatus is in progress
+                val drawStatusInProgress = e.drawStatus == DrawStatus.IN_PROGRESS
+                // show ProgressBar if MapView is drawing and lock modeTV
+                progressBar.visibility = if (drawStatusInProgress) View.VISIBLE else View.GONE
+                modeTV.isEnabled = !drawStatusInProgress
```

Comment:
I don't know that it's helpful to disable to mode button during drawing. It causes the mode to flash and I'm not sure there's not reason to block it's functionality while drawing's happening

Reply:
Regarding disabling the mode button, I noticed the Java [sample](https://github.com/Esri/arcgis-runtime-samples-java/blob/85501bb51ad7b7aa8a020d7ec80e40fdb0c292e7/feature_layers/toggle-between-feature-request-modes/src/main/java/com/esri/samples/toggle_between_feature_request_modes/ToggleBetweenFeatureRequestModesSample.java#L135) using it which could be helpful.  If while loading a lot of features it would be easy to halt the app while constantly switching between FeatureRequestModes while they are being loaded.

Testing it on by removing the `DrawStatusChangedListener` the sample works fine as long as too many features are not being loaded at once. I think that should be okay for the sample and I've gone ahead and made the change, but let me know what you would think. @TADraeseke